common/src/sql/dbinit.sql

```diff
@@ -1419,6 +1419,98 @@ CREATE INDEX ON omicron.public.update_available_artifact (
     targets_role_version
 );
 
+/*
+ * System updates
+ */
+CREATE TABLE omicron.public.system_update (
+    /* Unique semver version */
+    version STRING(64) PRIMARY KEY,
+
+    -- The ID is not strictly necessary here because we're using the version
+    -- string as the PK. However, removing it would make this one of the only
+    -- resources without an ID, which means we'd have to do a bunch of special
+    -- stuff that we get more or less for free with other resources. For
+    -- example, paginating by ID (note this is distinct from sorting by ID).
+
+    /* Identity metadata (asset) */
+    id UUID NOT NULL,
+    time_created TIMESTAMPTZ NOT NULL,
+    time_modified TIMESTAMPTZ NOT NULL
+);
```

Comment:
Using the version as the PK while still keeping the ID around is one of the stranger parts of this PR, but I landed on it after trying less weird-sounding alternatives that turned out to be uglier.

Reply:
Okay, minor nitpick, that doesn't really matter much but might help with consistency:

1. Can we make the ID the `PRIMARY KEY` if we're gonna have it?
2. Can we then create a unique index on the `version`?

I *think* this should be functionally very similar, but it matches the pattern of other objects with "both IDs and unique names".

Reply:
So the problem here was with the lookup macros. I could not figure out how to get a nice lookup-by-version with `lookup_resource!` without making `version` the primary key, because the way the macro is designed, it sounds like it has to be either by primary key or by name, and a non-PK version string is neither. I would love to find out that I understood this wrong, or that it's not that hard to change the macro (I looked but got spooked).

https://github.com/oxidecomputer/omicron/pull/2100/files#diff-6e518912406e7c5a535066ef28084be22fdbdae61895bab565eb656aa4682e01R698-R705